### PR TITLE
Authorization - Add roleassignmentmetrics calls and fix correctness issues

### PR DESCRIPTION
**Pull request by jdhms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

**Comment by jdhms** [comment]
> Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

@chiragg4u Thank you, fixed

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json
Comment (filizt) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:399]: >"500" [](start = 10, length = 5)

Curious why you defined 500 response model. Was there a requirement?

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2019-08-01-preview/authorization-UsageMetricsCalls.json
Comment (filizt) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2019-08-01-preview/authorization-UsageMetricsCalls.json:5]: >"version": "2019-08-01-preview" [](start = 4, length = 31)

This seems like a very old API version. Is there a reason for checking it in the swagger repo now?

**Comment by filizt** [comment]
There are breaking changes in the PR that has to go through the breaking changes review. Please find more information on the process here: https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json
Comment (jdhms) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:399]: In some cases we return a message in a format that is not compatible with the default response schema

**Comment by jdhms** [comment]
> There are breaking changes in the PR that has to go through the breaking changes review. Please find more information on the process here: https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes

This PR was opened during the window to fix broken for KPIs and its purpose is to fix s360 issues, would [this](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/405/Window-to-Fix-Broken) still apply?

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2019-08-01-preview/authorization-UsageMetricsCalls.json
Comment (jdhms) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2019-08-01-preview/authorization-UsageMetricsCalls.json:5]: It's to fix s360 issues during the window to fix broken, this preview version is still in use and was flagged for not being documented

**Comment by filizt** [comment]
> > There are breaking changes in the PR that has to go through the breaking changes review. Please find more information on the process here: https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes
> 
> This PR was opened during the window to fix broken for KPIs and its purpose is to fix s360 issues, would [this](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/405/Window-to-Fix-Broken) still apply?

It makes sense to me however you need to bring it up with API Breaking Changes person who is reviewing this PR.

**Comment by jdhms** [comment]
> It makes sense to me however you need to bring it up with API Breaking Changes person who is reviewing this PR.

@filizt will do, thanks! do we need anything else for ARM sign-off prior to then?

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2019-08-01-preview/examples/RoleAssignmentMetrics_GetForSubscription.json
Comment (filizt) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2019-08-01-preview/examples/RoleAssignmentMetrics_GetForSubscription.json:4]: >api-version": "2020-03-01-preview" [](start = 5, length = 34)

Conflicts with API version of the file path

**Comment by filizt** [comment]
> > It makes sense to me however you need to bring it up with API Breaking Changes person who is reviewing this PR.
> 
> @filizt will do, thanks! do we need anything else for ARM sign-off prior to then?

Just added a comment for a nit issue. Other than that LGTM. Will have to wait until the breaking changes board review to finalize from ARM side.

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**
